pubsub.go

```diff
@@ -598,14 +666,19 @@ func (p *PubSub) GetTopics() []string {
 // Publish publishes data under the given topic
 func (p *PubSub) Publish(topic string, data []byte) error {
 	seqno := p.nextSeqno()
-	p.publish <- &Message{
-		&pb.Message{
-			Data:     data,
-			TopicIDs: []string{topic},
-			From:     []byte(p.host.ID()),
-			Seqno:    seqno,
-		},
+	m := &pb.Message{
+		Data:     data,
+		TopicIDs: []string{topic},
+		From:     []byte(p.host.ID()),
```

Comment:
This needs to be the ID of the key we're using to sign.

Reply:
it's also not a bug in the current implementation, as we don't allow arbitrary keys yet.

Reply:
I will fix it anyway, so that it's futureproof (in case we want to allow arbitrary keys in the future)

Reply:
done.